src/test/java/org/jenkinsci/plugins/ewm/acceptance/ExternalWorkspaceManagerPluginTest.java

```diff
@@ -0,0 +1,170 @@
+                        "   } \n" +
+                        "} \n" +
+                        "node ('test') { \n" +
+                        "   exws (extWorkspace) { \n" +
+                        "       def content = readFile(file: 'marker') \n" +
+                        "       if (content != 'content') error('Content mismatch: ' + content) \n" +
+                        "   } \n" +
+                        "}", DISK_POOL_ID));
+
+        Build build = job.startBuild();
+        build.shouldSucceed();
+        assertThat(build.getConsole(), containsString(String.format("Running in %s/%s/%s", fakeNodeMountingPoint, job.name, build.getNumber())));
+
+        verifyExternalWorkspacesAction(job.name, build);
+    }
+
+    @Test
+    public void shareWorkspaceTwoJobsTwoNodes() {
+        WorkflowJob upstreamJob = createWorkflowJob(String.format(
+                "def extWorkspace = exwsAllocate '%s' \n" +
+                        "node ('linux') { \n" +
+                        "   exws (extWorkspace) { \n" +
+                        "      writeFile file: 'marker', text: 'content' \n " +
+                        "   } \n" +
+                        "}", DISK_POOL_ID));
+
+        Build upstreamBuild = upstreamJob.startBuild();
+        upstreamBuild.shouldSucceed();
+        assertThat(upstreamBuild.getConsole(), containsString(String.format("Running in %s/%s/%s", fakeNodeMountingPoint, upstreamJob.name, upstreamBuild.getNumber())));
+        verifyExternalWorkspacesAction(upstreamJob.name, upstreamBuild);
+
+        WorkflowJob downstreamJob = createWorkflowJob(String.format("" +
+                "def run = selectRun '%s' \n" +
+                "def extWorkspace = exwsAllocate selectedRun: run \n" +
+                "node ('test') { \n" +
+                "   exws (extWorkspace) { \n" +
+                "       def content = readFile(file: 'marker') \n" +
+                "       if (content != 'content') error('Content mismatch: ' + content) \n" +
+                "   } \n" +
+                "}", upstreamJob.name));
+
+        Build downstreamBuild = downstreamJob.startBuild();
+        downstreamBuild.shouldSucceed();
+        assertThat(downstreamBuild.getConsole(), containsString(String.format("Running in %s/%s/%s", fakeNodeMountingPoint, upstreamJob.name, upstreamBuild.getNumber())));
+        verifyExternalWorkspacesAction(upstreamJob.name, downstreamBuild);
+    }
+
+    @Test
+    public void externalWorkspaceCleanup() {
+        WorkflowJob job = createWorkflowJob(String.format("" +
+                "def extWorkspace = exwsAllocate '%s' \n" +
+                "node ('linux') { \n" +
+                "	exws (extWorkspace) { \n" +
+                "		try { \n" +
+                "           writeFile file: 'foobar.txt', text: 'any' \n" +
+                "		} finally { \n" +
+                "			step ([$class: 'WsCleanup']) \n" +
+                "		} \n" +
```

Comment:
Hidden tabs break indentation.

Reply:
Hmm indeed. I should use spaces instead, right?